metrics/bin/metrics.dart

```diff
@@ -0,0 +1,163 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:crypto/crypto.dart';
+import 'package:logging/logging.dart';
+import 'package:metrics/metrics.dart';
+import 'package:metrics/src/options.dart';
+import 'package:path/path.dart' as path;
+import 'package:pubspec_parse/pubspec_parse.dart';
+
+Future<void> main(List<String> arguments) async {
+  late Options options;
+  bool displayUsage = false;
+  try {
+    options = parseOptions(arguments);
+    displayUsage = options.help;
+  } catch (e) {
+    displayUsage = true;
+    exitCode = 64; // EX_USAGE (by convention, see sysexits.h)
+  }
+  if (displayUsage) {
+    print('dart run metrics [arguments]');
+    print(usage);
+    return;
+  }
+
+  final pubspecPath = path.join(path.current, 'pubspec.yaml');
+  final pubspec = Pubspec.parse(await File(pubspecPath).readAsString());
+
+  hierarchicalLoggingEnabled = true;
+  _log.level = options.verbose ? Level.INFO : Level.WARNING;
+
+  // Ensure realm_generator has run (EXPENSIVE!)
+  // Not really needed currently, as we don't pick up features yet,
+  // but it ensures the realm_generator has been run
+  final process = await Process.start('dart', [
+    'run',
+    'build_runner',
+    'build',
+    '--delete-conflicting-outputs',
+  ]);
+  await stdout.addStream(process.stdout);
+
+  if (Platform.environment['CI'] != null ||
+      Platform.environment['REALM_DISABLE_ANALYTICS'] != null) {
+    _log.info('Skipping metrics upload');
+    return;
+  }
+
+  try {
+    await uploadMetrics(options, pubspec);
+  } catch (e, s) {
+    _log.warning('Failed to upload metrics', e, s);
+    // We don't set exitCode > 0 on runtime errors!
+    // This script is called during build (via gradle, podspec, etc.)
+    // and we don't want to be the cause of a broken build!
+  }
+  exit(0); // why is this needed?
+}
+
+Future<void> uploadMetrics(Options options, Pubspec pubspec) async {
+  final hostId = await machineId();
+
+  final metrics = await generateMetrics(
+    distinctId: hostId,
+    targetOsType: options.targetOsType,
+    targetOsVersion: options.targetOsVersion,
+    anonymizedMacAddress:
+        hostId, // cannot get this with dart, using hostId instead :-/ (similar to realm-js)
```

Comment:
we could parse the getmac comamnd output on Windows and ifconfig  on *nix. The question is do we need to do that.

Reply:
```suggestion
        hostId,
```